File: typescript/src/prime-numbers.ts
function isPrimeNumber(num: number) {
	if (num <= 1) return false

	const firstHalf = Math.round(num / 2)
	const secondHalf = num - firstHalf

	let currentFirst = firstHalf <= 1 ? num + 1 : 0
	let currentSecond = secondHalf <= 1 ? num + 1 : 0
	let index = 2

	while(currentFirst < num || currentSecond < num) {
		if (currentFirst < num) {
			// Update currentFirst
			currentFirst = firstHalf * index
		}

		if (currentSecond < num) {
			// Update second
			currentSecond = secondHalf * index
		}

		index += 1
	}

	return (currentFirst === num || currentSecond === num) ? false : true
}

console.log(
	isPrimeNumber(97) // true
)
// console.log(
// 	isPrimeNumber(1) // false
// )

// console.log(
// 	isPrimeNumber(2) // true
// )

// console.log(
// 	isPrimeNumber(3) // true
// )

// console.log(
// 	isPrimeNumber(4) // false
// )

// console.log(
// 	isPrimeNumber(5) // true
// )

File: python/src/prime_numbers.py
def is_prime_number(num: int):
	if num <= 1:
		return False

	first_half = num // 2
	second_half = num - first_half

	current_first = num + 1 if first_half <= 1 else 0
	current_second = num + 1 if second_half <= 1 else 0

	index = 2

	while current_first < num or current_second < num:
		if current_first < num:
			current_first = first_half * index

		if current_second < num:
			current_second = second_half * index

		index += 1

	return False if current_first == num or current_second == num else True

if __name__ == "__main__":
	print(
		is_prime_number(97) # true
	)

	print(
		is_prime_number(1) # false
	)

	print(
		is_prime_number(2) # true
	)

	print(
		is_prime_number(3) # true
	)

	print(
		is_prime_number(4) # false
	)

	print(
		is_prime_number(5) # true
	)